### PR TITLE
add checks for outcome=None

### PR DESCRIPTION
**Pull request by wallacbe**

### CODE REVIEW

**Comment by wallacbe** [comment]
https://github.com/google/openhtf/issues/556

File: examples/all_the_things.py [hunk 1]
```diff
@@ -95,12 +95,19 @@ def attachments(test):
   test.attach_from_file('example_attachment.txt')
 
 
+@htf.TestPhase(run_if=lambda: False)
```
Comment (fahhem) [inline examples/all_the_things.py]: add a test for this please

File: openhtf/core/phase_executor.py [hunk 2]
```diff
@@ -199,9 +199,10 @@ def execute_phase(self, phase):
     while not self._stopping.is_set():
       outcome = self._execute_phase_once(phase)
```
Comment (fahhem) [inline openhtf/core/phase_executor.py]: make this return `PhaseOutcome(openhtf.PhaseResult.CONTINUE)` and undo everything else

Comment (grybmadsci) [inline openhtf/core/phase_executor.py]: I think line 217 should return `PhaseOutcome(openhtf.PhaseResult.CONTINUE)` instead, returning None there breaks the calling convention of `_execute_phase_once()`.

File: examples/all_the_things.py (hunk 1 above)
Comment (wallacbe) [inline examples/all_the_things.py]: Added test in exe_test

**Comment by wallacbe** [comment]
Made changes

**Pull request merged**